tox.ini

```diff
@@ -14,14 +14,18 @@ deps =
 
 [testenv:unit]
 commands =
-    pytest -vv tests -k operator
-
-[testenv:selenium]
-commands =
-    pytest -vv tests -k selenium
+    pytest -vv tests/unit
 
 [testenv:lint]
 commands =
     flake8 {toxinidir}/src {toxinidir}/tests
     black --check {toxinidir}/src {toxinidir}/tests
 
+[testenv:integration]
+deps =
+    juju
+    pytest
+    pytest-operator
+    selenium
+    selenium-wire
```

Comment:
Feels a little redundant with `test-requirements.txt`, but I realize you can't put things like `pytest-operator` into `test-requirements.txt` otherwise it'll break your unit tests... Really all options feel crappy for this and likely this is fine :)

Reply:
Yes, I agree, I personally feel like we should just keep all the dependencies defined in tox.

Reply:
I got back and forth, so I'm curious to hear your reasons.  I hated them being in tox because making a dev environment was annoying until I found out about `tox --devenv`, now I'm not sure which I like better.  

My thoughts are:
Having them in tox:
* + Feels well integrated with tox
* + Easy granular control (lint tests don't need pytest, unit cannot have pytest-operator, etc etc)
* - could make the tox file pretty cluttered (not so much for our current cases though)
Having them outside tox:
* + might be easier for people who aren't familiar with tox?
* + managing with something like poetry allows for automated package management and pinning

They all feel pretty subjective except for the last one - the automated management of package updates feels compelling